l10n.py

```diff
@@ -48,235 +63,325 @@
-                    return codecs.open(f, 'r', 'utf-8')
-                except:
-                    print_exc()
+                    return open(f, 'r', encoding='utf-8')
+
+                except Exception:
+                    logger.exception(f'could not open {f}')
+
             return None
-        elif getattr(sys, 'frozen', False) and platform=='darwin':
-            return codecs.open(join(self.respath(), '%s.lproj' % lang, 'Localizable.strings'), 'r', 'utf-16')
-        else:
-            return codecs.open(join(self.respath(), '%s.strings' % lang), 'r', 'utf-8')
 
+        elif getattr(sys, 'frozen', False) and platform == 'darwin':
+            return (self.respath() / f'{lang}.lproj' / 'Localizable.strings').open('r', encoding='utf-16')
 
-class Locale(object):
+        return (self.respath() / f'{lang}.strings').open('r', encoding='utf-8')
 
-    def __init__(self):
-        if platform=='darwin':
+
+class _Locale:
+    """Locale holds a few utility methods to convert data to and from localized versions."""
+
+    def __init__(self) -> None:
+        if platform == 'darwin':
             self.int_formatter = NSNumberFormatter.alloc().init()
             self.int_formatter.setNumberStyle_(NSNumberFormatterDecimalStyle)
             self.float_formatter = NSNumberFormatter.alloc().init()
             self.float_formatter.setNumberStyle_(NSNumberFormatterDecimalStyle)
             self.float_formatter.setMinimumFractionDigits_(5)
             self.float_formatter.setMaximumFractionDigits_(5)
 
-    def stringFromNumber(self, number, decimals=None):
-        # Uses the current system locale, irrespective of language choice.
-        # Unless `decimals` is specified, the number will be formatted with 5 decimal
-        # places if the input is a float, or none if the input is an int.
+    def stringFromNumber(self, number: Union[float, int], decimals: int = None) -> str:  # noqa: N802
+        warnings.warn(DeprecationWarning('use _Locale.string_from_number instead.'))
+        return self.stringFromNumber(number, decimals)
+
+    def numberFromString(self, string: str) -> Union[int, float, None]:  # noqa: N802
+        warnings.warn(DeprecationWarning('use _Locale.number_from_string instead.'))
+        return self.numberFromString(string)
+
+    def preferredLanguages(self) -> Iterable[str]:  # noqa: N802
+        warnings.warn(DeprecationWarning('use _Locale.preferred_languages instead.'))
+        return self.preferred_languages()
+
+    def string_from_number(self, number: Union[float, int], decimals: int = None) -> str:
+        """
+        Convert a number to a string.
+
+        Uses the current system locale, irrespective of language choice.
+
+        :param number: The number to stringify
+        :param decimals: The number of decimals to return, defaults to 5 if the given number is a float, otherwise None
+        :return: the stringified number
+        """
+        decimals = decimals if decimals is not None else 5
```

Comment:
I'm struggling to see how this is different from just having `decimals: int =  5` in the function signature.

1. decimals not passed, so it defaults to None, and ends up set to 5 here.
2. decimals passed as None, it will be set to 5 here.
3. decimals passed as a valid integer, it will be set to that here.

I guess this means that if you literally want to have no decimals on what *is* a floating point number you'd need to pass `0`.  Indeed the next bit of code handles that.

Reply:
Yeah I think that was a "I translated from or-magic to a ternary without looking to closely". Fixed now